qa/build.gradle

```diff
@@ -0,0 +1,174 @@
+    api "org.apache.logging.log4j:log4j-core:${versions.log4j}"
+    api "junit:junit:${versions.junit}"
+    testImplementation "org.opensearch.test:framework:${opensearch_version}"
+    testImplementation(testFixtures(rootProject))
+}
+
+ext {
+    licenseFile = rootProject.file('LICENSE.txt')
+    noticeFile = rootProject.file('NOTICE.txt')
+}
+
+def tmp_dir = project.file('build/private/artifact_tmp').absoluteFile
+tmp_dir.mkdirs()
+String default_bwc_version = System.getProperty("bwc.version")
+String neural_search_bwc_version = System.getProperty("tests.bwc.version", default_bwc_version)
+boolean isSnapshot = neural_search_bwc_version.contains("-SNAPSHOT")
+String neural_search_bwc_version_no_qualifier = isSnapshot ? neural_search_bwc_version - "-SNAPSHOT" : neural_search_bwc_version
+
+String os_platform = "linux"
+String artifact_type = "tar"
+String file_ext = "tar.gz"
+
+if (Os.isFamily(Os.FAMILY_WINDOWS)) {
+    os_platform = "windows"
+    artifact_type = "zip"
+    file_ext = "zip"
+}
+
+task deletetempDirectories {
+    doFirst {
+        File[] tempFiles = tmp_dir.listFiles()
+        for (File child : tempFiles) {
+            if (child.exists() && child.toString().contains("opensearch-")) {
+                Files.delete(child.toPath());
+            }
+        }
+    }
+}
+
+// Task to pull opensearch artifact from archive
+task pullOpensearchArtifact {
+   dependsOn "deletetempDirectories"
+
+   doLast{
+        ext{
+            if (isSnapshot) {
+                srcUrl = "https://ci.opensearch.org/ci/dbc/distribution-build-opensearch/${neural_search_bwc_version_no_qualifier}/latest/${os_platform}/x64/${artifact_type}/dist/opensearch/opensearch-${neural_search_bwc_version_no_qualifier}-${os_platform}-x64.${file_ext}"
+            } else {
+                srcUrl = "https://artifacts.opensearch.org/releases/bundle/opensearch/${neural_search_bwc_version}/opensearch-${neural_search_bwc_version}-${os_platform}-x64.${file_ext}"
+            }
+        }
+        ant.get(
+                src: srcUrl,
+                dest: tmp_dir.absolutePath,
+                httpusecaches: false
+        )
+        copy {
+            if (Os.isFamily(Os.FAMILY_WINDOWS)) {
+                from zipTree(java.nio.file.Path.of(tmp_dir.absolutePath, "opensearch-${neural_search_bwc_version_no_qualifier}-${os_platform}-x64.${file_ext}"))
+            } else {
+                from tarTree(java.nio.file.Path.of(tmp_dir.absolutePath, "opensearch-${neural_search_bwc_version_no_qualifier}-${os_platform}-x64.${file_ext}"))
+            }
+            into tmp_dir.absolutePath
+        }
+   }
+}
+
+// Task to pull ml plugin from archive
+task pullMlCommonsBwcPlugin {
+    doLast {
+        copy {
+            from(java.nio.file.Path.of(tmp_dir.absolutePath, "opensearch-${neural_search_bwc_version_no_qualifier}", "plugins", "opensearch-ml"))
+            into java.nio.file.Path.of(tmp_dir.absolutePath, "opensearch-ml")
+        }
```

Comment:
Should we delete the file like we do for pullBwcPlugin?

Reply:
In pullBWCPlugin it deletes opensearch artifact.

The execution flow is
1) pullOpensearchArtifact 
2) Pull knn plugin
3) pull ml plugin
4) pull neural plugin
5) Delete opensearch artifact

So the reason it is kept in pullBwc plugin is because it the last one to be called. Therefore it deletes it after the use.

To delete individual package we use
https://github.com/opensearch-project/neural-search/pull/515/files#diff-07e82fba5bdc43702a0da2ef075f370622dd8ca817b16329d33fa12b9816a2e5R142